chromium_src/third_party/blink/renderer/core/page/create_window.cc

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/core/page/create_window.h"
+
+#include "brave/third_party/blink/renderer/farbling/brave_session_cache.h"
+#include "third_party/blink/public/web/web_window_features.h"
+#include "third_party/blink/renderer/core/execution_context/execution_context.h"
+#include "third_party/blink/renderer/core/frame/local_dom_window.h"
+#include "third_party/blink/renderer/core/frame/screen.h"
+#include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
+
+// Because we are spoofing screenX and screenY, we need to offset the position
+// when a page script opens a new window in screen coordinates. And because
+// we are spoofing screen width and height, we should artificially limit the
+// window size to that width and height as well, so that window.open can't
+// be used to probe the screen size.
+
+namespace blink {
```

Comment:
why is this in blink instead of anonymous?

Reply:
fixed